### ISSUE TITLE
Server in debug mode won't start client unless server starts with specific current directory

### ISSUE THREAD

**Issue by BentTranberg** [fsbolero/Template#16]
This is just a tip for what I believe would be an improvement.

The template solution with server has these lines in Startup.fs :

```
#if DEBUG
            .AddHotReload(templateDir = "../TdWebServer.Client")
#endif
```

When I try to start the server outside of VS, it's quite natural for me to CD into the server's `bin\Debug\netcoreapp3.0` and start the executable there. That won't work because the templateDir is a relative path of the current directory. When run inside VS, the current directory is that of the server project's directory.

IMHO an application must not rely on the current directory for any purpose other than letting a user of a desktop application or command line application navigate disks and directories, and open files the user want to work with. Files and directories that are relative to an application's executable must be located by other means than the current directory. A server should never use the current directory.

I am not that familiar with web programming, and don't know whether it is common practice to use current directory in any way in various web server platforms.

There's also another line that I don't know enough about, but it works :

```
    let books =
        Path.Combine(env.ContentRootPath, "data/books.json")
        |> File.ReadAllText
```

I have fixed the problem in my project by setting templateDir as follows. It will then be relative to the executable rather than the current directory. I can then start the server regardless of where my current directory is. The books are also still displayed, so I assume `env.ContentRootPath` is not relative to the current directory.

```
#if DEBUG
            .AddHotReload(templateDir = IO.Path.Combine(AppContext.BaseDirectory, "../../../../TdWebServer.Client"))
#endif
```

**Comment by Tarmil** [comment]
Good point, this path should indeed be made independent from the working directory. I think it would be even more robust to use:

```fsharp
.AddHotReload(templateDir = Path.Combine(__SOURCE_DIRECTORY__, "../TdWebServer.Client")
```

This will still work if you ever change the build output directory for whatever reason.

> There's also another line that I don't know enough about, but it works :
> 
> ```
>     let books =
>         Path.Combine(env.ContentRootPath, "data/books.json")
>         |> File.ReadAllText
> ```

That's because `.json` files are marked `CopyToOutputDirectory` in ASP.NET Core projects. [By default, `env.ContentRootPath` is the directory containing the app assembly](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-2.2#content-root), and if you look in your bin folder, you'll see that `data/books.json` was copied there.

**Comment by BentTranberg** [comment]
Absolutely agree with you. I wanted something like that, but didn't know how, and didn't know it was that easy. Also, thanks for the information about env.ContentRootPath. Very helpful to me.

**Comment by Tarmil** [comment]
Closing; the templateDir was fixed in v0.8.
